File: Java/1539. Kth Missing Positive Number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int l = 0, r = arr.length - 1, m;

        while (l <= r) {
            m = (r - l)/2 + l;
            if (arr[m] - m - 1 < k) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return l + k;
    }
}

File: Python3/1539. Kth Missing Positive Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        start = arr[0] - 1
        if start >= k:
            return k

        k -= start
        prev = arr[0]

        for n in arr[1:]:
            diff = n - prev - 1
            if k <= diff:
                return prev + k

            k -= diff
            prev = n

        return arr[-1] + k